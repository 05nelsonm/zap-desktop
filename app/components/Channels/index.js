export ChannelCapacity from './ChannelCapacity'
export ChannelMoreButton from './ChannelMoreButton'
export ChannelStatus from './ChannelStatus'
export ChannelSummaryList from './ChannelSummaryList'
export ChannelSummaryListItem from './ChannelSummaryListItem'
