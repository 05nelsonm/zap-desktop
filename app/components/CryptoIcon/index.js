import CryptoIcon from './CryptoIcon'

export default CryptoIcon
