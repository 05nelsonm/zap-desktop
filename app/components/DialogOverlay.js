import styled from 'styled-components'
import { Flex } from 'rebass'

const Overlay = styled(Flex)`
  position: absolute;
  z-index: 99999;
  background-color: rgba(255, 255, 255, 0.25);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

export default Overlay
