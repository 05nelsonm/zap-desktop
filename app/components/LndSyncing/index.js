import LndSyncing from './LndSyncing'

export default LndSyncing
