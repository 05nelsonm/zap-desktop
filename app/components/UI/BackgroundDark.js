import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundDark
 * @example
 * <BackgroundDark />
 */
class BackgroundDark extends React.Component {
  render() {
    return <Box bg="darkestBackground" color="primaryText" {...this.props} />
  }
}

export default BackgroundDark
