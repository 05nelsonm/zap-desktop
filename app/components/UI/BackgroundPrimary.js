import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundPrimary
 * @example
 * <BackgroundPrimary />
 */
class BackgroundPrimary extends React.Component {
  render() {
    return <Box bg="primaryColor" color="primaryText" {...this.props} />
  }
}

export default BackgroundPrimary
