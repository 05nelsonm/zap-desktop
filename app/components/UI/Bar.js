import styled from 'styled-components'
import { variant, space, height } from 'styled-system'

const bars = variant({
  key: 'bars',
})

const Bar = styled.hr`
  height: 1px;
  margin: 0;
  padding: 0;
  border: 0;
  background-color: ${props => props.theme.colors.primaryText};
  ${space}
  ${height}
  ${bars}
`

Bar.defaultProps = {
  variant: 'normal',
}

Bar.displayName = 'Bar'

export default Bar
