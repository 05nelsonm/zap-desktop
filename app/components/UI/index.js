export ActionBar from './ActionBar'
export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export Button from './Button'
export Checkbox from './Checkbox'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export Notification from './Notification'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Value from './Value'
export Wizard from './Wizard'
