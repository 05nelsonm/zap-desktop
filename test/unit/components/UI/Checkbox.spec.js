import React from 'react'
import renderer from 'react-test-renderer'
import { Checkbox } from 'components/UI'
import { Form } from 'informed'
import { dark } from 'themes'
import { ThemeProvider } from 'styled-components'

describe('component.UI.Checkbox', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={dark}>
          <Form>
            <Checkbox field="name" label="cb" />
            <Checkbox field="name2" label="cb" description="desc" />
          </Form>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
