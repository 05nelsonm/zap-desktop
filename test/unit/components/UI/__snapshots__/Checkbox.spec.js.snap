// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Checkbox should render correctly 1`] = `
.c4 {
  margin-left: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  width: 18px;
  color: #242633;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 18px;
  border-radius: 5px;
  border: 1px solid #959595;
  background-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover .c2 {
  border-color: #fd9800;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c5 {
  font-size: 13px;
  color: #ffffff;
  line-height: 1.4;
}

.c6 {
  margin-top: 8px;
  font-size: 13px;
  color: #959595;
  line-height: 1.4;
}

<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    checked={false}
    className="c0"
    disabled={false}
    onClick={[Function]}
  >
    <input
      checked={false}
      className="c1"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      checked={false}
      className="c2 c3"
      color="primaryColor"
      disabled={false}
      width={18}
    />
    <div
      className="c4"
    >
      <div
        className="c5"
        color="primaryText"
        fontSize="m"
      >
        cb
      </div>
    </div>
  </div>
  <div
    checked={false}
    className="c0"
    disabled={false}
    onClick={[Function]}
  >
    <input
      checked={false}
      className="c1"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      checked={false}
      className="c2 c3"
      color="primaryColor"
      disabled={false}
      width={18}
    />
    <div
      className="c4"
    >
      <div
        className="c5"
        color="primaryText"
        fontSize="m"
      >
        cb
      </div>
      <div
        className="c6"
        color="gray"
        fontSize="m"
      >
        desc
      </div>
    </div>
  </div>
</form>
`;
